sdk/storage/src/common/storage_credential.cpp

```diff
@@ -35,11 +35,63 @@ namespace Azure { namespace Storage {
 
       if (!key.empty() || !value.empty())
       {
-        result[std::move(key)] = std::move(value);
+        connectionStringMap[std::move(key)] = std::move(value);
       }
     }
 
-    return result;
-  }
+    auto getWithDefault = [](const std::map<std::string, std::string>& m,
+                             const std::string& key,
+                             const std::string& defaultValue = std::string()) {
+      auto ite = m.find(key);
+      return ite == m.end() ? defaultValue : ite->second;
+    };
+
+    ConnectionStringParts connectionStringParts;
+
+    std::string defaultEndpointsProtocol
+        = getWithDefault(connectionStringMap, "DefaultEndpointsProtocol", "https");
+    std::string EndpointSuffix
+        = getWithDefault(connectionStringMap, "EndpointSuffix", ".core.windows.net");
+    std::string accountName = getWithDefault(connectionStringMap, "AccountName");
+    if (accountName.empty())
```

Comment:
As talked offline, unless the account name needs to be used in constructing the URL, it should not be mandatory.

Reply:
fixed